Extension/src/common.ts

```diff
@@ -318,19 +349,19 @@ export function resolveVariables(input: string, additionalEnvironment: {[key: st
         ret = ret.replace(regexp(), (match: string, ignored1: string, varType: string, ignored2: string, name: string) => {
             // Historically, if the variable didn't have anything before the "." or ":"
             // it was assumed to be an environment variable
-            if (varType === undefined) {
+            if (!varType) {
                 varType = "env";
             }
-            let newValue: string;
+            let newValue: string | undefined;
             switch (varType) {
                 case "env": {
-                    let v: string | string[] = additionalEnvironment[name];
+                    let v: string | string[] = additionalEnvironment2[name];
                     if (isString(v)) {
                         newValue = v;
                     } else if (input === match && isArrayOfString(v)) {
                         newValue = v.join(";");
                     }
-                    if (!isString(newValue)) {
```

Comment:
why is this changing?  `isString` should be able to remove the undefined typing.

Reply:
newValue will only be either undefined, or a string.  Use of isString() here was superfluous.  Removed it for simplicity.